src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/ITagHelperComponent.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Razor.TagHelpers
+{
+    /// <summary>
+    /// Contract used to filter matching HTML elements.
```

Comment:
I wouldn't explain the difference between the two. I'd explain that `ITagHelperComponent` enables users to mutate the representation of an HTML element. ITagHelper on the other hand is still a contract to filter matching HTML elements.